src/runtime/current_thread/runtime.rs

```diff
@@ -42,12 +44,32 @@ impl Runtime {
         // And now put a single-threaded executor on top of the timer. When there are no futures ready
         // to do something, it'll let the timer or the reactor to generate some new stimuli for the
         // futures to continue in their life.
-        let executor = CurrentThread::new_with_park(timer);
+        let mut executor = CurrentThread::new_with_park(timer);
 
-        let runtime = Runtime { reactor_handle, timer_handle, executor };
+        // Create an mpsc channel for scheduling new tasks on this runtime
+        // from a different thread
+        // XXX: Should this use a bounded channel with an arbitrary limit instead, and then
+        // panic spawning if the channel is full?
+        let (spawn_sender, spawn_receiver) = mpsc::unbounded();
+        executor.spawn(spawn_receiver.for_each(|future| {
```

Comment:
We *might* want to do this only the first time that `handle()` is called, but it complicates things slightly. Opinions?